.github/workflows/foundry-gas-diff.yml

```diff
@@ -0,0 +1,61 @@
+name: Report gas diff
+
+env:
+  RPC_URL_MAINNET: ${{ secrets.RPC_URL_MAINNET }}
+  RPC_URL_GOERLI: ${{ secrets.RPC_URL_GOERLI }}
+  RPC_URL_OPTIMISM: ${{ secrets.RPC_URL_OPTIMISM }}
+  RPC_URL_ARBITRUM: ${{ secrets.RPC_URL_ARBITRUM }}
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    # Optionally configure to run only for changes in specific files. For example:
+    # paths:
+    # - src/**
+    # - test/**
+    # - foundry.toml
+    # - remappings.txt
+    # - .github/workflows/foundry-gas-diff.yml
+
+jobs:
+  compare_gas_reports:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: 'core'
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+
+      - name: Install Foundry
+        uses: onbjerg/foundry-toolchain@v1
+        with:
+          version: nightly
+
+      # Add any step generating a gas report to a temporary file named gasreport.ansi. For example:
+      - name: Run tests
+        run: ./snapshot.sh > ../gasreport.ansi # <- this file name should be unique in your repository!
+        env:
+          # make fuzzing semi-deterministic to avoid noisy gas cost estimation
+          # due to non-deterministic fuzzing (but still use pseudo-random fuzzing seeds)
+          FOUNDRY_FUZZ_SEED: 0x${{ github.event.pull_request.base.sha || github.sha }}
+
+      - name: Compare gas reports
+        uses: Rubilmax/foundry-gas-diff@v3.16
```

Comment:
On the other hand, it looks like this expects a different format (the one from `forge test --gas-report`) instead of the one from `forge snapshot`. So you may need to revert to what you had before, but with an added `--match-contract ".*Gas"` flag)

i.e. in the "Run tests" step, use `forge test --gas-report --match-contract ".*Gas" > ../gasreport.ansi`